draft-ietf-jsonpath-base.md

```diff
@@ -1502,10 +1502,22 @@ The well-typedness of function expressions can now be defined in terms of this t
 
 For a function expression to be well typed:
 
-1. its declared type must be well typed in the context in which it occurs, and
-2. its arguments must be well typed for the declared type of the corresponding parameters.
+1. its function name MUST be registered (see {{iana-fnex}}),
+2. its declared type must be well typed in the context in which it occurs, and
+3. its arguments must be well typed for the declared type of the corresponding parameters.
 
-(1) As per the grammar, a function expression can occur in three different
+(1) The registered set of function extensions is the source of knowledge regarding the declared return and
+parameter types of a function extension.
+
+Although a function expression being treated as well typed implies its function name is registered,
+an implementation need not treat all registered functions as well typed.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+If an implementation does not support a function extension used in a query, it MUST treat the
+function extension as not well typed.
```

Comment:
See my previous comment: The error message is allowed to say the function extension is not supported, so "as not well typed" is overstating it.

Reply:
Sorry - I've lost track of your previous comment.

The reason I say  it MUST treat the function extension as not well typed is so that it's clear that section 2.1 applies (a query that is not well typed MUST raise an error).

A helpful implementation will provide diagnostics as part of the error - in this case, an error message saying the function extension is not supported. Do we have to spell that out or can we leave that as a quality of implementation consideration?

Reply:
Ah, found your previous comment https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/commit/e25a509952eeae5c66facd5fd98a073c844b1b20#r109974852. Let me try to iron that out.

Reply:
See bdf27be40ee2f5d5995dcdd6a13d7944952b3869.